ydb/core/persqueue/pq_impl.cpp

```diff
@@ -2345,6 +2464,94 @@ void TPersQueue::HandleSplitMessageGroupRequest(ui64 responseCookie, const TActo
     ctx.Send(partActor, new TEvPQ::TEvSplitMessageGroup(responseCookie, std::move(deregistrations), std::move(registrations)));
 }
 
+void TPersQueue::HandleGetOwnershipRequestForShadowPartition(const ui64 responseCookie,
+                                                             const NKikimrClient::TPersQueuePartitionRequest& req,
+                                                             const TActorId& sender,
+                                                             const TActorContext& ctx)
+{
+    Y_ABORT_UNLESS(req.HasWriteId());
+
+    ui64 writeId = req.GetWriteId();
+    ui32 partitionId = req.GetPartition();
+
+    if (TxWrites.contains(writeId) && TxWrites[writeId].Partitions.contains(partitionId)) {
+        ui32 shadowPartitionId = TxWrites[writeId].Partitions[partitionId];
+        Y_ABORT_UNLESS(ShadowPartitions.contains(shadowPartitionId));
+        TPartitionInfo& partition = ShadowPartitions.at(shadowPartitionId);
+
+        if (partition.InitDone) {
+            const TActorId& actorId = partition.Actor;
+            TActorId pipeClient = ActorIdFromProto(req.GetPipeClient());
+
+            HandleGetOwnershipRequest(responseCookie, actorId, req, ctx, pipeClient, sender);
+        } else {
+            partition.GetOwnershipRequests.emplace_back(responseCookie, req, sender);
+        }
+    } else {
+        GetOwnershipRequests.push_back({responseCookie, req, sender});
```

Comment:
Fill map TWrites/ShadowPartitionsId here. Create shadow partition actor only when map writeid/partitionid->shadow partition id is stored. In this case GetOwnershipRequests queue will be needed only in partition.

Reply:
done
https://github.com/ydb-platform/ydb/pull/1037/commits/74c832f44de40a1c11abdeb95e1b95dd7266a9bc